Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -138,6 +142,19 @@ u16 TMDReader::GetGroupId() const
   return Common::swap16(m_bytes.data() + offsetof(TMDHeader, group_id));
 }
 
+std::string TMDReader::GetGameID() const
+{
+  char game_id[6];
+  std::memcpy(game_id, m_bytes.data() + offsetof(TMDHeader, title_id) + 4, 4);
+  std::memcpy(game_id + 4, m_bytes.data() + offsetof(TMDHeader, group_id), 2);
+
+  for (char c : game_id)
```

Comment:
This can utilize `all_of`.

```cpp
...
  const bool all_printable = std::all_of(std::begin(game_id), std::end(game_id), [](char c) {
    return std::isprint(c, std::locale::classic());
  });
  
  if (all_printable)
    return std::string(game_id, sizeof(game_id));

  return StringFromFormat("%016" PRIx64, GetTitleId());
}
```